compose/config/config.py

```diff
@@ -199,14 +238,52 @@ def load(config_details):
 
     Return a fully interpolated, extended and validated configuration.
     """
+    version = get_config_version(config_details)
+    if version not in COMPOSEFILE_VERSIONS:
+        raise ConfigurationError('Invalid config version provided: {0}'.format(version))
```

Comment:
I was doing this before but I think that moving it to the else was too late for some reason. I'm gonna test it again.